[release-4.18] Fix TestAnalyzeAnomaliesTool_Description test failure

## Problem

CI is failing on the `release-4.18` branch due to test failure in `TestAnalyzeAnomaliesTool_Description`.

**Latest Failed CI Run**: https://github.com/tosin2013/openshift-cluster-health-mcp/actions/runs/21036962284

**Error**:
```
Error: "..." does not contain "label selector"
--- FAIL: TestAnalyzeAnomaliesTool_Description (0.00s)
```

## Root Cause

Test expects "label selector" (with space) but the description uses "label_selector" (with underscore).

- **Test expectation** (`internal/tools/analyze_anomalies_test.go:22`):
  ```go
  assert.Contains(t, desc, "label selector")  // expects space
  ```
- **Actual description** (`internal/tools/analyze_anomalies.go:55`):
  ```go
  "- label_selector: Filter by Kubernetes labels..."  // uses underscore
  ```

## Proposed Fix

**Option 1 (Recommended)**: Update test to expect "label_selector"
```go
// File: internal/tools/analyze_anomalies_test.go:22
assert.Contains(t, desc, "label_selector")
```

**Option 2**: Update description to say "label selector" (less preferred due to inconsistency with JSON field name)

## Impact

- ❌ All CI runs failing on `release-4.18`
- ❌ Blocks releases for OpenShift 4.18 compatibility
- ❌ Prevents bug fixes from being released for 4.18

## Related

- This issue exists across all branches: `main`, `release-4.18`, `release-4.19`, `release-4.20`
- See related issues for tracking per-branch

## Tasks

- [ ] Decide on fix approach (Option 1 or 2)
- [ ] Apply fix to `release-4.18` branch
- [ ] Verify CI passes
- [ ] Tag release if needed

--- a/internal/tools/analyze_anomalies.go
+++ b/internal/tools/analyze_anomalies.go
@@ -52,7 +52,7 @@ FILTERING OPTIONS:
 - namespace: Scope to a specific namespace
 - deployment: Analyze specific deployment (mutually exclusive with pod)
 - pod: Analyze specific pod (mutually exclusive with deployment)
-- label_selector: Filter by Kubernetes labels (e.g., 'app=flask')
+- label selector: Filter by Kubernetes labels (e.g., 'app=flask')
 
 Example questions this tool answers:
 - "Are there any anomalies in CPU usage?"

--- a/internal/tools/analyze_anomalies_test.go
+++ b/internal/tools/analyze_anomalies_test.go
@@ -19,7 +19,7 @@ func TestAnalyzeAnomaliesTool_Description(t *testing.T) {
 	assert.Contains(t, desc, "anomalies")
 	assert.Contains(t, desc, "deployment")
 	assert.Contains(t, desc, "pod")
-	assert.Contains(t, desc, "label_selector")
+	assert.Contains(t, desc, "label selector")
 }
 
 func TestAnalyzeAnomaliesTool_InputSchema(t *testing.T) {

